js/views/conversation_view.js

```diff
@@ -201,11 +201,45 @@
             this.$('.bottom-bar form').addClass('active');
         },
 
+        updateUnread: function() {
+            this.updateLastSeenIndicator();
+            this.model.markRead();
+        },
+
         onOpened: function() {
             this.view.resetScrollPosition();
             this.$el.trigger('force-resize');
             this.focusMessageField();
-            this.model.markRead();
+
+            if (this.inProgressFetch) {
+                this.inProgressFetch.then(this.updateUnread.bind(this));
+            } else {
+                this.updateUnread();
+            }
```

Comment:
An alternative to this pattern would be to initially set `this.inProgressFetch = Promise.resolve()`. Then you don't need to nullify and check for null because you always have a promise to chain on to.